apps/balancer-tools/src/app/apr/api/route.ts

```diff
@@ -241,14 +265,23 @@ export async function GET(request: NextRequest) {
   const order = (searchParams.get("order") as Order) || "desc";
   const limit = parseInt(searchParams.get("limit") ?? "0") || Infinity;
   const offset = parseInt(searchParams.get("offset") ?? "0");
-
   let responseData;
 
+  try {
+    validateSearchParams(poolId, roundId);
+  } catch (error) {
+    return NextResponse.json(
+      { error: (error as Error).message },
+      { status: 400 },
+    );
+  }
+
   if (poolId && roundId) {
     return NextResponse.json(
       await getDataFromCacheOrCompute(
         `pool_${poolId}_round_${roundId}`,
         async () => fetchDataForPoolIdRoundId(poolId, roundId),
+        parseInt(roundId) === parseInt(Round.currentRound().value),
```

Comment:
🥇

Reply:
👌

Reply:
This goes directly against clean code's [Eliminate Boolean Arguments](https://www.informit.com/articles/article.aspx?p=1392524), I'm ok with using it here since it's not changing the function's output

Reply:
hmm... that makes sense. Perhaps a better implementation would pass an empty cache key and simply avoid caching altogether when the cacheKey evaluates to nullish?